main/input_default.cpp

```diff
@@ -1338,11 +1338,14 @@ String InputDefault::get_joy_button_string(int p_button) {
 
 int InputDefault::get_joy_button_index_from_string(String p_button) {
 	for (int i = 0; i < JOY_BUTTON_MAX; i++) {
-		if (p_button == _buttons[i]) {
+		if (_buttons[i] == nullptr) {
+			break;
+		}
+		if (p_button == String(_buttons[i])) {
 			return i;
 		}
 	}
-	ERR_FAIL_V(-1);
+	ERR_FAIL_V_MSG(-1, vformat(RTR("Could not find a button index matching the string \"%s\"."), p_button));
```

Comment:
We don't translate error messages so this shouldn't use `RTR()`. (my suggestion was using C++ R-strings to avoid having to escape `"`, but it's fine with he escape too)

Reply:
Ah,, I misunderstood your suggestion. Ok I applied it.